app-shells/squirrelsh/files/squirrelsh-1.2.7-gcc6.patch

```diff
@@ -0,0 +1,15 @@
+--- a/shell/base.cpp
++++ b/shell/base.cpp
+@@ -14,10 +14,11 @@
+ // You should have received a copy of the GNU General Public License
+ // along with this program. If not, see <http://www.gnu.org/licenses/>.
+ 
+-#include "common.h"
+ #include <string.h>
+ #include <string>
+ 
++#include "common.h"
```

Comment:
I generally try to keep my patches to a minimum to enable the code's original pre-GCC6 intent without necessarily using more **proper** code constructs that may have more wide-ranging consequences throughout.

In the case of app-shells/squirrelsh, there exists in `${S}/common.h`:
```
#if !defined(min)
#	define min(a, b) ((a) < (b) ? (a) : (b))
#endif
```
This macro name forcibly hides all overloaded declarations of `std::min` (such as `std::min(const T& a, const T& b, Compare comp)`).  Specifically, the problem happens because there is a call to `std::min(const T& a, const T& b, Compare comp)` in `/usr/lib/gcc/x86_64-pc-linux-gnu/6.3.0/include/g++-v6/bits/stl_algobase.h` and that declaration has been hidden by the presence of a macro with the same name.

However, testing what would happen if I were to change that code in `${S}/common.h` to:
```
#include <algorithm>
using std::min;
```
I get errors like:
```
hash_adler32.cpp:90:50: error: no matching function for call to 'min(SQInteger&, int)'
   size_t r = fread(block, 1, min(left, BLOCK_SIZE), file);
                                                  ^
```
because `std::min` places more stringent demands on its arguments (such as requiring them to be the exact same type).  It would require explicit casts in more files.  At that point, I'd prefer to let upstream decide how to deal with it.